### DISCUSSION TITLE
Consider making "scan for quests here" higher?

### DISCUSSION THREAD

**Discussion by matkoniecz** [streetcomplete/StreetComplete#4130] [Feedback]
It is used far, far more often than enabling team mode so making it easier to click would be nice.

**Comment by mnalis** [comment]
I agree, especially now as `team mode` is demoted to another line of text, so it is **_now much easier to misclick_** `team mode` when one wanted `scan now`. 

Or just move `team mode` to `Settings` menu (that would have a benefit that muscle memory for often used `scan` would stay the same as it did until now; i.e. only `team mode` icon would be replaced by `overlay` icon)? 

Because, one needs to click `team mode` much **less often** than `change preset` (maybe few times a month), and `presets` (which users using them might be switching at least few times a day or more frequently) are burried deep into `Settings`.

**Comment by HolgerJeromin** [comment]
"nobody" will find team mode if hidden in settings.,..

**Reply by mnalis** [reply]
Wouldn't they? It seems that 2/3 of the users have found a way to enable "disabled by default" quests, which are also hidden in the Settings. i.e. https://www.openstreetmap.org/user/wielandb/diary/399164. 

But there are also other options, like:
- making the hamburger open icon panel sized 2x3 instead of 2x2, so `Team mode` can be icon again (and we then even have one free place, for example for [quick changing](https://github.com/streetcomplete/StreetComplete/issues/3827) active `Preset`!)
- leaving `Scan for quests here` at bottom of the menu (where it was before) and moving `Enter team mode` line at the top, above the icons.
- moving `Scan for quests here` to the main screen - it is at least as important (and as often used, if not more - especially in fast moving vehicleslike bus, car, train - where autodownload doesn't work good) as the `GPS` button...
- some other idea for making  `Scan for quests here` easier to hit and harder to miss (e.g. click something else similarly-looking instead) 

But current situation in 45.0-alpha1 is highly problematic, as the **_very often used_**  `Scan for quests here` is not being clicked correctly (instead hitting `Enter team mode`) -- both because the line is smaller than icon, and because there is other similar line very near it, and because muscle memory. 

Just today I misclicked `team mode` (instead of `Scan for quests here`) more than dozen times , and it was a relatively short mapping  :cry:

**Reply by matkoniecz** [reply]
> It seems that 2/3 of the users have found a way to enable "disabled by default" quests, which are also hidden in the Settings. i.e. https://www.openstreetmap.org/user/wielandb/diary/399164.

I suspect that they count also all users who solve roof shape quest when it was enabled, just because it is disabled now.

**Reply by mnalis** [reply]
Perhaps someone should run a modified script to account for that, then. But still, according to that page, `AddRoofShape` accounted for only about `6%` of quests solved, so it maybe does not change that figure drastically?

**Comment by legofahrrad** [comment]
> making the hamburger open icon panel sized 2x3 instead of 2x2, so Team mode can be icon again

I think that'll be a good option, it could look like this then:
![Screenshot_Overlay_Menu_new](https://user-images.githubusercontent.com/92865620/174659543-ca8525db-8e76-467d-bde9-b6f4009ce464.jpg)
Maybe the "scan for quests here" should be on the right side, as the ratio of left-/right-hander seems to be, that most people are right-hander. (so swap "team-mode" and "scan quests" buttons)

**Reply by mnalis** [reply]
Yeah, maybe it might work. My main worry is if that would function on smaller screens.
 
If somebody with smaller older phone might take a screenshot of how that menu looks now (in 2x2 form), then we might estimate if it 2x3 would also fit.

And I'm also somewhat fond of that full-width `Scan for quests here` because (as you note) it is equally accessible to both left-handed and right-handed people, which is IMHO important for such frequently used feature (especially on modern bigger screens, where one cannot reach left end of the screen with right hand (or vice-versa) especially when using phone one-handed.

**Reply by Helium314** [reply]
Due to the ways Android UI things work, it should not look much different for older phones. Actually the screenshot/mockup might come from an old device (from an age before the screens became ridiculously tall), judging from the image dimensions.

On my S4 mini, which qualifies as somewhat old, there is enough space for the dialog to double in height (but I have no navigation bar).

Otherwise I totally agree with this idea, plus with @mnalis' suggestion of switching team mode and scan.

**Reply by HolgerJeromin** [reply]
This one would be an optimized right hand layout:

|  profile | setting  |
|---|---|
| about  | team  |
|  layer|  scan  |

**Reply by mnalis** [reply]
I'm trying that now here https://github.com/mnalis/StreetComplete/tree/ui-hamburger-2x3 (`.apk` at https://github.com/mnalis/StreetComplete/actions/runs/2539041269 if someone wants to try latest). I've done that 2x3 grid that @HolgerJeromin has shown, but _also_ currently left additional  (wide) button to `Scan for quests` - until I can decide how usable using only square button is.  

![small_Screenshot_20220622_022637_de westnordost streetcomplete mn debug](https://user-images.githubusercontent.com/156656/174918354-132f0f17-21d2-4efd-ad2b-de051ca5e285.jpg)

However, unless `@westnordost` changes his mind :cry:  https://github.com/streetcomplete/StreetComplete/discussions/4113#discussioncomment-2994882 it seems it will likely only end up in https://github.com/Helium314/StreetComplete fork.

**Reply by mnalis** [reply]
I've done  9a12aa3075c068f82131752612d44d33427f8aed ([.apk](https://github.com/mnalis/StreetComplete/actions/runs/2545964934)) which:
- preserves ordering from previous releases (MyProfile / Team Mode, Settings / About, and then add Overlays / Scan below) as my muscle memory was clicking setting at wrong place; so probably better to keep it in familiar places
- removes long rectangle `Scan for quests here` so I can check how it work only with square (when testing I was clicking on long rectangle most of the times).
![small_Screenshot_20220623_024229_de westnordost streetcomplete mn debug](https://user-images.githubusercontent.com/156656/175184056-ab1cee27-e430-4f22-87d3-7f7e7cbc8133.jpg)

**Comment by mnalis** [comment]
BTW @Helium314 when testing that menu changes, I've noticed that this build of mine based on latest upstream [master](4ae8fcec6305f6d653d9f6ae91e51e6b8ebc2f2f) (https://github.com/mnalis/StreetComplete/actions/runs/2545964934) when refreshing the quests in preset which only has same quests enabled it has a flash where it shows all the quests, and then hides them (wherefore before it didn't happen). I guess it could be related to your recent work on caching / performance improvements (rather then my simple menu UI changes), so just a heads-up.

Here are videos comparing the two:

https://user-images.githubusercontent.com/156656/175185222-06a426cd-09a0-4a0b-90ac-19ece5bc28f5.mp4

https://user-images.githubusercontent.com/156656/175185220-2463e1e2-2b4d-475d-8b2a-c4e5698f2adf.mp4
